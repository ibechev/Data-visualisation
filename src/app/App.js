import React from "react";

import Alerts from "./components/alerts/Alerts";
import ClientsInfo from "./components/clientsInfo/ClientsInfo";

const App = () => {
  return (
    <div className="app">
      <ClientsInfo />
      <Alerts />
    </div>
  );
};

export default App;
