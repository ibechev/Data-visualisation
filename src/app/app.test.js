import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import App from "./App";

describe("Expect <App />", () => {
  it("to render correctly", () => {
    const wrapper = renderer.create(<App />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("to render one <ClientsInfo /> component", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find("ClientsInfo")).toHaveLength(1);
  });

  it("to render one <Alerts /> component", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find("Alerts")).toHaveLength(1);
  });
});
