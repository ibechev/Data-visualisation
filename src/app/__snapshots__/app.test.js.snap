// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <App /> to render correctly 1`] = `
<div
  className="app"
>
  <section
    className="clients-info"
  >
    <header
      className="clients-info-header"
    >
      <button
        className="button no-body"
        onClick={[Function]}
        type="button"
      >
        Clients
        <i
          className="fas fa-caret-right"
        />
      </button>
      <button
        className="button no-body"
        onClick={[Function]}
        type="button"
      >
        <i
          className="fas fa-cog"
        />
      </button>
    </header>
    <div
      className="clients-info-charts"
    >
      <div
        className="loading-spinner"
      >
        <i
          className="fas fa-spinner fa-spin"
        />
        <p>
          Loading 
          information
           ...
        </p>
      </div>
    </div>
  </section>
  <section
    className="alerts"
  >
    <header
      className="alerts-header"
    >
      <div
        className="alerts-info"
      >
        <h4
          className="alerts-title"
        >
          Alerts
        </h4>
      </div>
      <div
        className="alerts-controls"
      >
        <div
          className="main-alerts-controls"
        >
          <ul
            className="controls-list"
          >
            <li
              className="control-li"
            >
              <button
                className="button no-body"
                onClick={[Function]}
                type="button"
              >
                <i
                  className="fas fa-chart-pie"
                />
              </button>
            </li>
            <li
              className="control-li"
            >
              <button
                className="button no-body"
                onClick={[Function]}
                type="button"
              >
                <i
                  className="fas fa-filter"
                />
              </button>
            </li>
            <li
              className="control-li"
            >
              <button
                className="button no-body"
                onClick={[Function]}
                type="button"
              >
                <i
                  className="fas fa-undo-alt"
                />
              </button>
            </li>
            <li
              className="control-li"
            >
              <button
                className="button no-body"
                onClick={[Function]}
                type="button"
              >
                <i
                  className="fas fa-cog"
                />
              </button>
            </li>
          </ul>
          <button
            className="button primary"
            onClick={[Function]}
            type="button"
          >
            Save 
            <i
              className="fas fa-caret-down"
            />
          </button>
        </div>
      </div>
    </header>
    <div
      className="loading-spinner"
    >
      <i
        className="fas fa-spinner fa-spin"
      />
      <p>
        Loading 
        alerts
         ...
      </p>
    </div>
  </section>
</div>
`;
